build.sbt

```diff
@@ -452,8 +452,29 @@ lazy val zincApiInfo = (project in internalPath / "zinc-apiinfo")
   .settings(
     name := "zinc ApiInfo",
     crossScalaVersions := compilerBridgeTestScalaVersions,
-    relaxNon212,
+    compilerVersionDependentScalacOptions,
     mimaSettings,
+    mimaBinaryIssueFilters ++= {
+      import com.typesafe.tools.mima.core._
+      import com.typesafe.tools.mima.core.ProblemFilters._
+      Seq(
+         exclude[IncompatibleMethTypeProblem]("xsbt.api.HashAPI.hashTypeParameters"),
+         exclude[IncompatibleMethTypeProblem]("xsbt.api.HashAPI.hashAnnotations"),
+         exclude[IncompatibleMethTypeProblem]("xsbt.api.HashAPI.hashParameters"),
+         exclude[DirectMissingMethodProblem]("xsbt.api.HashAPI.hashDefinitionsWithExtraHashes"),
+         exclude[DirectMissingMethodProblem]("xsbt.api.HashAPI.hashSeq"),
+         exclude[IncompatibleMethTypeProblem]("xsbt.api.HashAPI.hashValueParameters"),
+         exclude[IncompatibleMethTypeProblem]("xsbt.api.HashAPI.hashAnnotationArguments"),
+         exclude[IncompatibleMethTypeProblem]("xsbt.api.HashAPI.hashTypes"),
+         exclude[IncompatibleMethTypeProblem]("xsbt.api.Visit.visitTypeParameters"),
+         exclude[IncompatibleMethTypeProblem]("xsbt.api.Visit.visitDefinitions"),
+         exclude[IncompatibleMethTypeProblem]("xsbt.api.Visit.visitAnnotationArguments"),
+         exclude[IncompatibleMethTypeProblem]("xsbt.api.Visit.visitAnnotations"),
+         exclude[IncompatibleMethTypeProblem]("xsbt.api.Visit.visitValueParameters"),
+         exclude[IncompatibleMethTypeProblem]("xsbt.api.Visit.visitParameters"),
+         exclude[IncompatibleMethTypeProblem]("xsbt.api.Visit.visitTypes"),
```

Comment:
the "api" in "xsbt.api" makes me concerned: is this public API?

Reply:
I think the "api" here refers the the functionality of this code (visiting the API).

It would be possible to leave deprecated methods in place with the old signatures if needed.